### ISSUE TITLE
Initial transition not created/callbacks not fired unless `state_machine` is touched

### ISSUE THREAD

**Issue by mikebaldry** [gocardless/statesman#547]
The documentation suggests implementing like:
```ruby
def state_machine
  @state_machine ||= OrderStateMachine.new(self, transition_class: OrderTransition,
                                                 association_name: :transitions,
                                                 initial_transition: true)
end
```

the initial transition is created and the callbacks then fired from this initializer, so if the `state_machine` isn't called after create of a model, this doesn't fire.

I'm not exactly sure of the scenarios `state_machine` isn't touched, if something happens internally with Statesman but not always or if it mostly happens that our code touches it in most cases, but we've had sporadic issues that has led me to add 
```ruby
after_create { state_machine }
```
to the models using Statesman, which appears to fix. 

Should we need to do this, if so, could it be added to the documentation, if not, any ideas where we're going wrong? 

Many thanks!

**Comment by chrise86** [comment]
~I'm also facing this issue but the `after_create` solution isn't working for me, at least not in a test environment. Is this still what you're doing @mikebaldry or did you come up with anything else?~

Scratch that, I _am_ facing the issue still, but that fix _does_ work for me. Turns out I'd installed the gem _just_ before the readme got updated, and so I'd installed using `~> 12.0.0` instead of `~>13.0.0` which introduced the `initial_transition` method itself.

Updating to `13.0.0` now allows me to create that initial transition, provided I manually initialise the state machine after create.

**TL;DR** is yes, this is an issue.

**Comment by chaadow** [comment]
yeah i noticed this, because all my index pages use a read-only replica ( with writes prevented) and i started getting 500 errors because statesman when accessing the state machine and it has no initial transition tries to insert that initial transition.

I manually monkey patched for my case, so that i check if i'm in a `reading` role with ActiveRecord, and skip that creation.

 but still, it confirms this issue

**Comment by jonleighton** [comment]
I also experienced this problem, which showed up as a deadlock in a test. I haven't fully understood why, but the workaround solves it anyway.
